### DISCUSSION TITLE
Option to padding image background on bestFit method

### DISCUSSION THREAD

**Discussion by ramtin2025** [claviska/SimpleImage#322] [Ideas]
When a set of images are produced using bestFit method, their size will be irregular and different. It would be better if there was an option for padding that would make all images the same size and cover the empty space with the selected color.

`->bestFit(800, 600, '#ffffff')`

**Comment by claviska** [comment]
While this is possible with the current API, it takes a few extra steps. I'm open to a new option (or method) that makes this easier, similar to what we're doing with [`thumbnail()`](https://github.com/claviska/SimpleImage#thumbnailwidth-height-anchor).

The `bestFit()` method is pretty well known already, so additional options should probably be an array.

```php
$image->bestFit(800, 600, [
  // When true, the resulting image will be the exact dimensions specified. 
  // Pixels that would normally be trimmed will be transparent by default, 
  // but can be changed with the `fill` option.
  'exact' => true,
  /* The fill color to use when `exact` is enabled. */
  'fill' => '#fff'
]);
```

When `$options['exact']` is true, the function would do this:

1. Call `bestFit($width, $height)` on the source image
2. Create a new image sized `$width` x `$height`
3. Fill the new image with `$options['fill']`
4. Overlay the best fit image onto the new image
5. Return the new image

How does that sound?

**Comment by jorban** [comment]
Is this something we can do now? or is this something in the works. I'm trying to make a thumbnail with a white background and rounded corners. If we can't do this now... what would it look like to take a logo, and add a white background with rounded corners? How would I code that properly using SimpleImage? Thanks :)

**Comment by ramtin2025** [comment]
I added these two methods in myImageClass to padding image with specified color:

    <?php
    
    class myImageClass extends \claviska\SimpleImage
    {
        public function resizeFitPadding($width, $height, $color = '#ffffff')
        {
            $frame = (new static())->fromNew($width, $height, $color);
            $overlay = $this->ntBestFit($width, $height);
            return $frame->overlay($overlay);
        }
    
        public function ntBestFit($maxWidth, $maxHeight) {
            // Calculate max width or height based on orientation
            if($this->getOrientation() === 'portrait') {
                $height = $maxHeight;
                $width = $maxHeight * $this->getAspectRatio();
            } else {
                $width = $maxWidth;
                $height = $maxWidth / $this->getAspectRatio();
            }
    
            // Reduce to max width
            if($width > $maxWidth) {
                $width = $maxWidth;
                $height = $width / $this->getAspectRatio();
            }
    
            // Reduce to max height
            if($height > $maxHeight) {
                $height = $maxHeight;
                $width = $height * $this->getAspectRatio();
            }
    
            return $this->resize($width, $height);
        }
    }
